src/item.cpp

```diff
@@ -2073,8 +2073,13 @@ bool Item::canDecay() const
 		return false;
 	}
 
-	const ItemType& it = Item::items[id];
-	if (getDecayTo() < 0 || it.decayTime == 0) {
+	uint32_t decayTime = getDuration();
+	if (decayTime == 0) {
+		const ItemType& it = Item::items[id];
+		decayTime = it.decayTime;
```

Comment:
add getDecayTime method to item class that would return it.decayTime?

Reply:
perfect!